### PR TITLE
Add LUA Synced read functions:

### PR DESCRIPTION
**Pull request by DeadnightWarrior**
Spring.GetUnitFeatureSeparation(unitID, featureID[, dist2D])
Spring.GetFeatureSeparation(featureID1, featureID2[, dist2D])

There is Spring.GetUnitSeparation(unitID1, unitID2[, dist2D]), this is the extension to features, and unit vs feature distances.

### CODE REVIEW

File: rts/Lua/LuaSyncedRead.cpp [hunk 1]
```diff
@@ -3711,6 +3713,36 @@ int LuaSyncedRead::GetUnitSeparation(lua_State* L)
...
+		return 0;
+	}
+	CFeature* feature = ParseFeature(L, __FUNCTION__, 2);
+	if (feature == NULL  || !IsFeatureVisible(L, feature)) {
+		return 0;
+	}
+
+	float3 pos1;
+	if (!IsAllyUnit(L, unit)) {
+		pos1 = unit->GetErrorPos(CLuaHandle::GetHandleReadAllyTeam(L));
```
Comment (ashdnazg) [inline rts/Lua/LuaSyncedRead.cpp]: pos1 won't be initialised if IsAllyUnit returns true.

**Comment by DeadnightWarrior** [comment]
I've copy/pasted the code for this functions from similar ones already present in LUA Sync Read
If this is wrong, then https://github.com/spring/spring/blob/develop/rts/Lua/LuaSyncedRead.cpp#L3688 is wrong too.

**Comment by ashdnazg** [comment]
Did you test if these functions actually work in the game?
You should have met errors before the fix if you had.

I ask because there was a similar case here https://github.com/spring/spring/pull/174 that could be avoided by testing.

**Comment by DeadnightWarrior** [comment]
Since Spring can't be compiled with VS2010 for quite some time, I have no way of testing it ingame before a dev build appears on Spring buildbot. Previous case was purely my fault since I oversaw a use case when testing it ingame.

**Comment by ashdnazg** [comment]
What about using mingw?
https://springrts.com/wiki/Building_Spring_on_Windows

**Comment by GoogleFrog** [comment]
I can build spring on windows. I followed that page.

**Pull request merged**